mongo/collection.go

```diff
@@ -545,7 +592,18 @@ func (coll *Collection) UpdateOne(ctx context.Context, filter interface{}, updat
 	return coll.updateOrReplace(ctx, f, update, false, rrOne, true, opts...)
 }
 
-// UpdateMany updates multiple documents in the collection.
+// UpdateMany executes an update operation (https://docs.mongodb.com/manual/reference/command/update/) to update
+// documents in the collection.
+//
+// The filter parameter must be a document containing query operators and can be used to select the documents to be
+// updated. It cannot be nil. An empty document (e.g. bson.D{}) should be used to select all documents. If the filter does
+// not match any documents, the operation will succeed and an UpdateResult with a MatchedCount of 0 will be returned.
```

Comment:
Perhaps more generally: "The Matched Count in the returned UpdateResult will equal the number of documents matching the filter, which may be zero if there were no matching documents."

Reply:
I think the exact description of `MatchedCount` belongs in the `UpdateResult` docs, not here.

Reply:
I'm not that concerned with redundancy.  It just seems weird to specify here the behavior of the less common case of MatchedCount of 0, but not the more common case of MatchedCount > 0.

Reply:
I called out `MatchedCount` of 0 specifically because I think a natural question is "what happens if no documents matched the filter" as that could be considered "success" or "error" from an API standpoint. If you don't think it's necessary, I'm open to removing that as well and leaving it as "If the filter does not match any documents, the operation will succeed."